src/cider/nrepl/middleware/info.clj

```diff
@@ -86,7 +87,28 @@
             see-also (clj-info/see-also (:ns var-info) (:name var-info))]
         (if (seq see-also)
           (merge {:see-also see-also} var-info)
-          var-info)))))
+          (if-let [p (:protocol var-info)]
+            (let [proto (deref p)
+                  x (filter #(extends? proto
+                                       %)
+                            (filter #(class? %)
+                                    (map val (mapcat ns-imports (all-ns)))))
+                  xs (map (fn [x]
+                            (let [xs (str/split (pr-str x)
+                                                #"\.")]
+                              (str (str/replace (str/join  "." (drop-last xs))
+                                                #"\_"
+                                                "-")
+                                   "/map->"
+                                   (last xs))))
+                          x)]
+              {:candidates (into {}
```

Comment:
is this returning `:candidates`? shouldn't it go in `compilment`?

Reply:
Those are different candidates. :-)

Reply:
I think we use the same key for several things here and there (I think it was used somewhere in `info` as well).